chain/store/splitstore/splitstore.go

```diff
@@ -28,8 +28,10 @@ var baseEpochKey = dstore.NewKey("baseEpoch")
 var log = logging.Logger("splitstore")
 
 type SplitStore struct {
-	baseEpoch abi.ChainEpoch
-	curTs     *types.TipSet
+	mx         sync.Mutex
```

Comment:
Let's switch this to atomic, it's close to 3x faster!

```
goos: darwin
goarch: amd64
pkg: github.com/testground/sdk-go
BenchmarkMutex
BenchmarkMutex-16          	59274573	        19.3 ns/op
BenchmarkAtomicInt64
BenchmarkAtomicInt64-16    	159052496	         7.66 ns/op
PASS
```

```go
package main

import (
	"sync"
	"sync/atomic"
	"testing"
)

func BenchmarkMutex(b *testing.B) {
	var mu sync.Mutex
	var v int64

	for i := 0; i < b.N; i++ {
		mu.Lock()
		v = int64(i)
		mu.Unlock()

		mu.Lock()
		_ = v
		mu.Unlock()
	}
}

func BenchmarkAtomicInt64(b *testing.B) {
	var v int64 // guarded by atomic

	for i := 0; i < b.N; i++ {
		atomic.StoreInt64(&v, int64(i))
		_ = atomic.LoadInt64(&v)
	}
}
```

Reply:
it doesn't really matter, this is a micro-optimization.

Reply:
also, we are accessing more complex things than integers, and things become a bit ugly with atomics.

Reply:
These are low-level operations, and a 3x speed up is worth, especially when the effort is close to none.

Reply:
That said, the position of this mutex threw me off. It is actually guarding `curTs` and `compacting`, not `baseEpoch`. If we were to use `atomic.Value` here, the speed-up is less noticeable.

```
goos: darwin
goarch: amd64
pkg: github.com/testground/sdk-go
BenchmarkMutex
BenchmarkMutex-16          	58329135	        19.1 ns/op
BenchmarkAtomicValue
BenchmarkAtomicValue-16    	98621289	        11.1 ns/op
PASS
```

If we need to modify these state variables atomically, then we are better off using locks. But if they're independent (like compacting is), we can use atomics.

Reply:
We still need the mutex for curTs, there is a little more complexity there. For compacting, I don't care particularly, it can be CAS.

Reply:
done.